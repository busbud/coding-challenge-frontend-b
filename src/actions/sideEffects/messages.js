export default {
	requestFailed: 'Request failed',
};
