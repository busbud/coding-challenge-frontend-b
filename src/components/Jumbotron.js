import React, { useContext } from 'react'
import Styled from 'styled-components'
import { AppContext } from '../contexts'

const Container = Styled.div`
  align-items: center;
  display: flex;
  justify-content: center;
  width: 100vw;
`

const Text = Styled.p`
  color: hotpink;
  font-size: 6vw;
  font-family: 'Permanent Marker', cursive;
`

export default () => {
  const { language } = useContext(AppContext)
  const text =
    language === 'EN' ? "Let's go to Osheaga!" : "Allons au festival d'Osheaga!"
  return (
    <Container>
      <Text>{text}</Text>
    </Container>
  )
}
