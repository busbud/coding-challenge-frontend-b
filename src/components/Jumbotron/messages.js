import { defineMessages } from 'react-intl';

export default defineMessages({
  title: {
    id: 'containers.jumbotron.title',
    defaultMessage: 'Find and compare cheap bus',
  },
});
