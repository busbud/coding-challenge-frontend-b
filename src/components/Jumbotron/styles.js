export default theme => ({
  root: {
    padding: `${theme.spacing.unit * 0.75}rem 0`,
  },
});
