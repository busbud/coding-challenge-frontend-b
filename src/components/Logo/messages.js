import { defineMessages } from 'react-intl';

export default defineMessages({
  alt: {
    id: 'components.logo.alt',
    defaultMessage: 'Busbud',
  },
});
