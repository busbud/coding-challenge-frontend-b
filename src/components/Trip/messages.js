import { defineMessages } from 'react-intl';

export default defineMessages({
  action_booking: {
    id: 'components.trip.action.booking',
    defaultMessage: 'Book now',
  },
  hours: {
    id: 'components.trip.hours',
    defaultMessage: '{amount}H ',
  },
  minutes: {
    id: 'components.trip.minutes',
    defaultMessage: '{amount}m',
  },
});
