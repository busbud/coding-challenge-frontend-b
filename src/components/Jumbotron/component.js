import React from 'react';
import { FormattedMessage } from 'react-intl';
import Grid from 'material-ui/Grid';
import Typography from 'material-ui/Typography';
import messages from './messages';
const propTypes = {};

const Component = ({ classes }) => (
  <Grid container justify="flex-start" className={classes.root}>
    <Grid item md={4} xs={12}>
      <Typography variant="display3" gutterBottom color="primary">
        <FormattedMessage {...messages.title} />
      </Typography>
    </Grid>
  </Grid>
);

Component.propTypes = propTypes;

export default Component;
