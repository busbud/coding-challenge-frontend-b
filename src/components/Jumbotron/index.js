import { withStyles } from 'material-ui/styles';
import styles from './styles';
import component from './component';

export default withStyles(styles)(component);
