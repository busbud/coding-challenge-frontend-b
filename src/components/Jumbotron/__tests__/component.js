import React from 'react';
import { shallow } from 'enzyme';
import { FormattedMessage } from 'react-intl';
import Typography from 'material-ui/Typography';
import Jumbotron from '../component';
import messages from '../messages';

const props = {
  classes: {
    root: 'fakeRootClass',
  },
};

describe('component | Jumbotron | component', () => {
  describe('render', () => {
    it('should render without throwing and error', () => {
      // given
      const wrapper = shallow(<Jumbotron {...props} />);

      // then
      expect(wrapper).toBeDefined();
    });

    it('should match snapshot', () => {
      // given
      const wrapper = shallow(<Jumbotron {...props} />);

      // then
      expect(wrapper).toMatchSnapshot();
    });
  });

  describe('Grid', () => {
    it('should have proper className props', () => {
      // given
      const wrapper = shallow(<Jumbotron {...props} />);
      const className = wrapper.props().className;

      // then
      expect(className).toBe(props.classes.root);
    });
  });
  describe('Typography', () => {
    it('should contains proper message', () => {
      // given
      const wrapper = shallow(<Jumbotron {...props} />);
      const formattedMessageProps = wrapper
        .find(Typography)
        .find(FormattedMessage)
        .props();

      // then
      expect(formattedMessageProps).toMatchObject(messages.title);
    });
  });
});
