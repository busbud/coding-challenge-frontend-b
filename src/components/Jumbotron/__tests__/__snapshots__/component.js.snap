// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`component | Jumbotron | component render should match snapshot 1`] = `
<WithStyles(GridWrapper)
  className="fakeRootClass"
  container={true}
  justify="flex-start"
>
  <WithStyles(GridWrapper)
    item={true}
    md={4}
    xs={12}
  >
    <WithStyles(Typography)
      color="primary"
      gutterBottom={true}
      variant="display3"
    >
      <FormattedMessage
        defaultMessage="Find and compare cheap bus"
        id="containers.jumbotron.title"
        values={Object {}}
      />
    </WithStyles(Typography)>
  </WithStyles(GridWrapper)>
</WithStyles(GridWrapper)>
`;
